libs/coin-framework/src/account/helpers.ts

```diff
@@ -0,0 +1,414 @@
+import { BigNumber } from "bignumber.js";
+import invariant from "invariant";
+import { getEnv } from "../env";
+import { encodeTokenAccountId } from "./accountId";
+import { emptyHistoryCache } from "./balanceHistoryCache";
+import type {
+  Account,
+  AccountLike,
+  AccountLikeArray,
+  SubAccount,
+  TokenAccount,
+  ChildAccount,
+} from "@ledgerhq/types-live";
+import {
+  CryptoCurrency,
+  TokenCurrency,
+  Unit,
+} from "@ledgerhq/types-cryptoassets";
+
+// by convention, a main account is the top level account
+// in case of an Account is the account itself
+// in case of a TokenAccount it's the parentAccount
+export const getMainAccount = (
+  account: AccountLike,
+  parentAccount?: Account | null | undefined
+): Account => {
+  const mainAccount = account.type === "Account" ? account : parentAccount;
+  invariant(mainAccount, "an account is expected");
+  return mainAccount as Account;
+};
+
+export const getAccountCurrency = (
+  account?: AccountLike
+): TokenCurrency | CryptoCurrency => {
+  switch (account?.type) {
+    case "Account":
+    case "ChildAccount":
+      return account.currency;
+
+    case "TokenAccount":
+      return account.token;
+
+    default:
+      throw new Error(
+        "invalid account.type=" + (account as unknown as { type: string })?.type
+      );
+  }
+};
+
+export const getAccountUnit = (account: AccountLike): Unit => {
+  switch (account.type) {
+    case "Account":
+      return account.unit;
+
+    case "TokenAccount":
+      return account.token.units[0];
+
+    case "ChildAccount":
+      return account.currency.units[0];
+
+    default:
+      throw new Error("invalid account.type=" + (account as AccountLike).type);
+  }
+};
+
+export const getAccountName = (account: AccountLike): string => {
+  switch (account.type) {
+    case "Account":
+    case "ChildAccount":
+      return account.name;
+
+    case "TokenAccount":
+      return account.token.name;
+
+    default:
+      throw new Error("invalid account.type=" + (account as AccountLike).type);
+  }
+};
+
+export const getAccountSpendableBalance = (account: AccountLike): BigNumber => {
+  switch (account.type) {
+    case "Account":
+    case "TokenAccount":
+      return account.spendableBalance;
+
+    case "ChildAccount":
+      return account.balance;
+
+    default:
+      throw new Error("invalid account.type=" + (account as AccountLike).type);
+  }
+};
+
+export const isAccountEmpty = (a: AccountLike): boolean => {
+  // if (a.type === "Account" && a.currency.family === "tron") {
+  //   const tronAcc = a as TronAccount;
+  //   // FIXME: here we compared a BigNumber to a number, would always return false
```

Comment:
this comment was actually wrong. `BigNumber("0").eq(0)` is true

Reply:
So you mean, I can safely remove the // FIXME comment?